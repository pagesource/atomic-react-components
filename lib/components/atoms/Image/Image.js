// @flow
import React from 'react';
import type { Node, ComponentType } from 'react';
import styled from 'styled-components';

import { getLabelFromChildren } from '../../../core/utils/elementLocator';
import styles from './Image.style';

type Props = {
  className: string,
  src: string,
  srcset?: string,
  sizes?: string,
  placeholderSrc?: string,
  alt: string,
  elementLocator?: string,
  inheritedStyles?: string,
  loading?: 'auto' | 'lazy' | 'eager',
  ref: () => void,
};

const Image = ({
  className,
  src,
  srcset,
  sizes,
  placeholderSrc,
  alt,
  inheritedStyles,
  ref,
  elementLocator,
  loading,
  ...other
}: Props): Node => (
  <img
    {...other}
    className={className}
    src={src}
    srcSet={srcset || null}
    sizes={sizes || null}
    alt={alt}
    ref={ref}
    loading={loading}
    data-locator={elementLocator || `image-${getLabelFromChildren(alt)}`}
    onError={
      placeholderSrc
        ? (event) => {
            // eslint-disable-next-line no-param-reassign
            event.target.src = placeholderSrc;
          }
        : null
    }
  />
);

Image.defaultProps = {
  srcset: '',
  sizes: '',
  placeholderSrc: '',
  inheritedStyles: '',
  loading: 'auto',
  elementLocator: '',
};

const StyledImage: ComponentType<Props> = styled(Image)`
  ${styles};
`;

export default StyledImage;

export { Image as ImageVanilla };

export type { Props as ImageProps };
