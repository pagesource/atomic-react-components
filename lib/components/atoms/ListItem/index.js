// @flow
export { default } from './ListItem';
export { ListItemVanilla } from './ListItem';
