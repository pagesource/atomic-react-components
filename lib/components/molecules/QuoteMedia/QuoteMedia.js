// @flow
/**
 *
 * QuoteMedia
 *
 */
import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';

import type { Node, ComponentType } from 'react';
import styles from './QuoteMedia.style';
import type { Props } from './types';

import Heading from '../../atoms/Heading/Heading';

const QuoteMedia = ({ className, title, quoteContent, HeadingType, children }: Props): Node => (
  <div className={classnames(className)}>
    {title && (
      <Heading HeadingType={HeadingType} className="title">
        {title}
      </Heading>
    )}
    <div className="container">
      {quoteContent && quoteContent()}
      <div className="media">{children}</div>
    </div>
  </div>
);

QuoteMedia.defaultProps = {
  className: 'quoteMedia',
  title: 'Anonymous',
  variation: 'column',
  HeadingType: 'h3',
};

const StyledQuoteMedia: ComponentType<Props> = styled(QuoteMedia)`
  ${styles};
`;

export default StyledQuoteMedia;

export { QuoteMedia as QuoteMediaVanilla };
