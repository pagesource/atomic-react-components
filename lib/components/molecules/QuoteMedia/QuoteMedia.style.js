import { css } from 'styled-components';

const styles = css`
  .title,
  .media {
    ${(props) => `
      padding: ${props.theme.spacing.unit2};
    `};
  }
  .container {
    display: flex;
    flex-direction: ${(props) => (props.variation === 'stacked' ? 'column' : 'row')};
  }
  ${(props) => (props.inheritedStyles ? props.inheritedStyles : '')};
`;
export default styles;
