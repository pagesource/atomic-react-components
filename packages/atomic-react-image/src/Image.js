import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { getLabelFromChildren } from '@xt-pagesource/atomic-react-utils';
import styles from './Image.style';

const Image = ({
  className,
  src,
  srcset,
  sizes,
  placeholderSrc,
  alt,
  inheritedStyles,
  ref,
  elementLocator,
  loading,
  ...other
}) => (
  <img
    {...other}
    className={className}
    src={src}
    srcSet={srcset || null}
    sizes={sizes || null}
    alt={alt}
    ref={ref}
    loading={loading}
    data-locator={elementLocator || `image-${getLabelFromChildren(alt)}`}
    onError={
      placeholderSrc
        ? (event) => {
            // eslint-disable-next-line no-param-reassign
            event.target.src = placeholderSrc;
          }
        : null
    }
  />
);

Image.propTypes = {
  className: PropTypes.string,
  src: PropTypes.string.isRequired,
  srcset: PropTypes.string,
  sizes: PropTypes.string,
  placeholderSrc: PropTypes.string,
  alt: PropTypes.string.isRequired,
  elementLocator: PropTypes.string,
  inheritedStyles: PropTypes.string,
  loading: PropTypes.oneOf(['auto', 'lazy', 'eager']),
  ref: PropTypes.func.isRequired,
};

Image.defaultProps = {
  className: '',
  srcset: '',
  sizes: '',
  placeholderSrc: '',
  inheritedStyles: '',
  loading: 'auto',
  elementLocator: '',
};

const StyledImage = styled(Image)`
  ${styles};
`;

export default StyledImage;

export { Image as ImageVanilla };
