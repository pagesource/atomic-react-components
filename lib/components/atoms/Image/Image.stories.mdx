import { Meta, Story, Preview, Props  } from '@storybook/addon-docs/blocks';
import Image, { ImageVanilla } from '.';
import defaultImage, { lazyLoadedImage, customImageLocator } from './Image.mock';

<Meta title="Atoms/Image" component={ImageVanilla} />


### Image Vanilla
<Preview>
  <Story name="Vanilla Image">
 <ImageVanilla {...defaultImage} />
  </Story>
</Preview>

### custom Image Locator
<Preview>
  <Story name="custom Image">
 <Image {...customImageLocator} />
  </Story>
</Preview>

### lazy Loaded Image
<Preview>
  <Story name="lazy Loaded">
 <Image {...lazyLoadedImage} />
  </Story>
</Preview>

## Props ðŸ”§

<Props of={ImageVanilla} />
