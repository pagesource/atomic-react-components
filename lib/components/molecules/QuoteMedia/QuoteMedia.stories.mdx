import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';

import Image from '../../atoms/Image/Image';
// Import Styled Component to showcase variations
import QuoteMedia, { QuoteMediaVanilla } from '.';
import defaultImage from '../../atoms/Image/Image.mock';
import { defaultQuoteMedia, stackedQuoteMedia } from './QuoteMedia.mock';

<Meta title="Molecules/Quote Media" component={QuoteMediaVanilla} />

### Knobs
<Preview>
  <Story name="Knobs">
<QuoteMediaVanilla {...defaultQuoteMedia} />
  </Story>
</Preview>

### Column
<Preview>
  <Story name="Column">
<QuoteMedia {...defaultQuoteMedia}>
      <Image {...defaultImage} />
    </QuoteMedia>
  </Story>
</Preview>

### Stacked
<Preview>
  <Story name="Stacked">
      <QuoteMedia {...stackedQuoteMedia}>
      <Image {...defaultImage} />
    </QuoteMedia>
  </Story>
</Preview>



