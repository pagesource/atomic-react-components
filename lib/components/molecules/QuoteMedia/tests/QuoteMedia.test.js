import React from 'react';

import { shallowWithTheme } from '../../../../../__mock__/themeMock';
import { QuoteMediaVanilla } from '../QuoteMedia';
import { ImageVanilla } from '../../../atoms/Image';
import defaultImage from '../../../atoms/Image/Image.mock';
import { defaultQuoteMedia } from '../QuoteMedia.mock';

describe('<QuoteMedia />', () => {
  let QuoteMediaComponent = '';
  beforeEach(() => {
    QuoteMediaComponent = shallowWithTheme(
      <QuoteMediaVanilla {...defaultQuoteMedia}>
        <ImageVanilla {...defaultImage} />
      </QuoteMediaVanilla>
    );
  });
  it('Should have component heading being displayed', () => {
    expect(QuoteMediaComponent.find('h2')).toBeTruthy();
    expect(QuoteMediaComponent.find('.title').text()).toBe(defaultQuoteMedia.title);
  });
  it('Should allow quote component passing', () => {
    expect(QuoteMediaComponent.props().quoteContent).not.toBeNull();
    expect(QuoteMediaComponent.find('.container').find('Blockquote')).toBeTruthy();
  });
  it('Should render media Image in component body', () => {
    expect(QuoteMediaComponent.props().children[0]).not.toBeNull();
    expect(QuoteMediaComponent.find('.media')).toBeTruthy();
    expect(QuoteMediaComponent.find('.container').find('.media').find('Image')).toBeTruthy();
  });
  test('Should render correctly', () => {
    expect(QuoteMediaComponent).toMatchSnapshot();
  });
});
