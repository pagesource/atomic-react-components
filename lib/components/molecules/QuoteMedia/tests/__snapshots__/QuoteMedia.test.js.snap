// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuoteMedia /> Should render correctly 1`] = `
<div
  className="quoteMedia"
>
  <Heading__StyledHeading
    HeadingType="h2"
    className="title"
  >
    Inspirational Quotes
  </Heading__StyledHeading>
  <div
    className="container"
  >
    <Blockquote__StyledBlockquote
      quoteAuthor="- Rear Admiral Grace Murray Hopper"
      quoteContent="Ask forgiveness, not permission."
    />
    <div
      className="media"
    >
      <Image
        alt="Dummy Image"
        elementLocator=""
        inheritedStyles=""
        loading="auto"
        placeholderSrc="/images/image-placeholder.png"
        sizes="(min-width: 36em) 33.3vw, 100vw"
        src="/images/dummies/320.png"
        srcset="/images/dummies/1280.png 1024w, /images/dummies/640.png 640w, /images/dummies/320.png 320w"
      />
    </div>
  </div>
</div>
`;
